src/common.js

```diff
@@ -0,0 +1,33 @@
+'use strict'
+/**
+ * The types of encoding states which can be defined for the success or failure from a request.
+ */
+const stateEncode = ['onSuccessEncode', 'onErrorEncode']
+
+/**
+ * Makes a buffer from a response stream.
+ *
+ * @param {stream} stream a response stream to read as a buffer
+ */
+const getBuffer = stream => new Promise((resolve, reject) => {
+  const parts = []
```

Comment:
let’s use the async generator API for the stream here so that we capture any errors this generates inside the promise. this is an oversight from the prior code, but it’s worth taking this on now since we’re doing so much to capture any error state and still throw the StatusError.

```js
const parts = []
for await (const buffer of stream) { 
  parts.push(stream)
}
return Buffer.concat(parts)
```